### PR TITLE
Fix duplicate code location error toasts + Add status filter to code locations page

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Recent changes to the WorkspaceContext where we split fetched by location caused us show the code location error toast multiple times. Once for each code location update after we've detected an errored code location. The fix is to instead fire the effect off of the errored code locations and preserve the reference if they haven't changed. 

Also adding a "state" filter to the code locations page for use in the error toast so that if you have 132 locations you can easily see which ones failed.

## How I Tested These Changes

Loaded an org with 132 code locations and saw a single error toast. Clicked on "view definitions" and saw it took me to the code location page with status = Failed as the filter.

Screenshot 2024-06-13 at 4 25 03 PM
Screenshot 2024-06-13 at 4 24 53 PM

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx
Comment (alangenfeld) [inline js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx:17]: whats the distinction between these two states exactly?

**Review by alangenfeld — COMMENTED** [review]
nice, this should be a good ux improvement

File: js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx:17]: Loading is the first time we see it. Just copying the logic from here: https://github.com/dagster-io/dagster/blob/a41c42e2981fd86c0fae66a73823db3acc942169/js_modules/dagster-ui/packages/ui-core/src/workspace/CodeLocationRowSet.tsx#L119

File: js_modules/dagster-ui/packages/ui-core/src/nav/useCodeLocationsStatus.tsx
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/nav/useCodeLocationsStatus.tsx:96]: I believe I can remove this, will do that before landing

File: js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx
Comment (alangenfeld) [reply js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx:17]: lost to phabricator history https://github.com/dagster-io/dagster/commit/c2056577761ca735eebea31eba030658e21459e4

**Review by alangenfeld — COMMENTED** [review]
[1] less confident on these filtered deep links, i think for loaded its likely the majority of the list and for updating/loading you might deep link to an empty list if the timing is off 

error / failed one is definitely a win

**Review by alangenfeld — APPROVED** [review]
code looks good tho, defer to you on making a call on [1]

**Pull request merged**
